enarx-keep-sgx-shim/src/heap.rs

```diff
@@ -200,3 +204,156 @@ impl Heap {
         0
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    const ERRNO_BASE: usize = !0xfff;
+
+    unsafe fn new_checked(bytes: &mut [u8]) -> Heap {
+        let aligned = slice(bytes.as_mut_ptr(), bytes.len())
+            .align_to_mut::<Page>()
+            .1;
+        let pages = aligned.len();
+        let span = Span {
+            start: aligned.as_mut_ptr() as _,
+            count: (aligned.len() * Page::size()) as _,
+        };
+        let heap = Heap::new(span);
+
+        // brk points to the first page
+        assert_eq!(heap.metadata.brk.start, heap.pages.as_ptr() as _);
+        assert_eq!(heap.metadata.brk.end, heap.pages.as_ptr() as _);
+
+        // exclude 2 pages for metadata and allocation map
+        assert_eq!(heap.pages.len(), pages - 2);
+
+        // no pages are allocated
+        for page in 0..heap.pages.len() {
+            assert!(!heap.is_allocated(page));
+        }
+
+        heap
+    }
+
+    fn oneshot(heap: &mut Heap, pages: usize) {
+        let brk_page = heap.pages.len() - pages;
+        let brk = heap.metadata.brk.start + brk_page * Page::size();
+        let ret = heap.brk(brk);
+        assert_eq!(ret, brk);
+
+        let steps = [
+            (Page::size(), 1),
+            (Page::size() / 2, 1),
+            (Page::size() + Page::size() / 2, 2),
+            (pages * Page::size(), pages),
+        ];
+
+        for (s, allocated) in steps.iter() {
+            let addr = heap.mmap(
+                0,
+                *s,
+                libc::PROT_READ as _,
+                (libc::MAP_PRIVATE | libc::MAP_ANONYMOUS) as _,
```

Comment:
There is a lot of wrapping in this file due to these two constants. Could you perhaps make some global `const`s?